### PR TITLE
ARROW-13872: [Java] ExtensionTypeVector does not work with RangeEqualsVisitor

### PR DESCRIPTION
**Pull request by BryanCutler**
This adds the ExtensionTypeVector to the VectorVisitor interface and implements that method for the existing visitors. Now when visitor like RangeEqualsVisitor that uses another ExtensionTypeVector, they can both use the visitor on the underlying storage vector.

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java [hunk 1]
```diff
@@ -214,6 +215,18 @@ public Boolean visit(NullVector left, Range range) {
     return true;
   }
 
+  @Override
+  public Boolean visit(ExtensionTypeVector<?> left, Range range) {
+    if (!validate(left) || !(right instanceof ExtensionTypeVector<?>)) {
```
Comment (liyafan82) [inline java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java:220]: For better performance, `!(right instanceof ExtensionTypeVector<?>)` could be placed first?

File: java/vector/src/test/java/org/apache/arrow/vector/types/pojo/TestExtensionType.java [hunk 2]
```diff
@@ -230,6 +236,48 @@ public void roundtripLocation() throws IOException {
     }
   }
 
+  @Test
+  public void testVectorCompare() {
+    ExtensionTypeRegistry.register(new UuidType());
```
Comment (liyafan82) [inline java/vector/src/test/java/org/apache/arrow/vector/types/pojo/TestExtensionType.java:241]: nit: can we reuse the `uuidType` object below?

**Comment by liyafan82** [comment]
@BryanCutler Thanks for working on this. 
Mostly LGTM, with a few minor comments. Please check if they are reasonable.

**Comment by BryanCutler** [comment]
Thanks for reviewing @liyafan82. Good catches with the comments, I updated to address them.

**Comment by liyafan82** [comment]
Merging. Thanks.

**Pull request closed without merging**